### PR TITLE
Add test cases for Service Selector

### PR DESCRIPTION
**Pull request by IrinaComindaru**
Related to [this issue](https://github.com/capsulajs/selector-service/issues/6)

### CODE REVIEW

File: tests/features/service-selector.feature [hunk 1]
```diff
@@ -0,0 +1,80 @@
+
+
+Scenario: Call setItemsStream with stream of items when Selector instance is empty
+    Given Selector Service with setItemsStream method
+    When  user calls setItemsStream with a stream of items
+    Then  all items are being loaded in the Selector instance
+    And   subscribing to items method returns the stored items
```
Comment (MikeSquall) [inline tests/features/service-selector.feature]:
```suggestion
    And   subscribing to items$ method returns the stored items
```

File: tests/features/service-selector.feature [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+Scenario: Call setItemsStream with a stream of new items when Selector instance is not empty
+    Given  Selector Service with setItemsStream method
+    And    several items are stored in the Selector instance
+    When   user calls setItemsStream with a stream of items that do not exist in the Selector instance
+    Then   the old items are replaced with the new items
+    And    subscribing to items method returns only the new stored ones
```
Comment (MikeSquall) [inline tests/features/service-selector.feature]:
```suggestion
    And    subscribing to items$ method returns only the new stored ones
```

File: tests/features/service-selector.feature [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+ Scenario: Call setItemsStream with a stream of items already stored in the Selector instance
```
Comment (MikeSquall) [inline tests/features/service-selector.feature]: This scenario shouldn't be merged with the previous one ?
The aim of of these 2 scenarios are the same: checking that something we input in the selector through `setItemsStream` is stored, no matter if it is a different stream or the same

File: tests/features/service-selector.feature [hunk 4]
```diff
@@ -0,0 +1,80 @@
+
+
+Scenario: Call setItemsStream with stream of items when Selector instance is empty
+    Given Selector Service with setItemsStream method
+    When  user calls setItemsStream with a stream of items
+    Then  all items are being loaded in the Selector instance
```
Comment (stephanebenayoun) [inline tests/features/service-selector.feature]: `Then  all items are being loaded in the Selector instance` this step is not needed, this is not something that will be checked.

File: tests/features/service-selector.feature [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    And   subscribing to items method returns the stored items
+
+Scenario: Call setItemsStream with a stream of new items when Selector instance is not empty
+    Given  Selector Service with setItemsStream method
+    And    several items are stored in the Selector instance
+    When   user calls setItemsStream with a stream of items that do not exist in the Selector instance
+    Then   the old items are replaced with the new items
```
Comment (stephanebenayoun) [inline tests/features/service-selector.feature]: Step not needed

File: tests/features/service-selector.feature [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+Scenario: Call setItemsStream with stream of items when Selector instance is empty
+    Given Selector Service with setItemsStream method
+    When  user calls setItemsStream with a stream of items
+    Then  all items are being loaded in the Selector instance
+    And   subscribing to items method returns the stored items
+
+Scenario: Call setItemsStream with a stream of new items when Selector instance is not empty
+    Given  Selector Service with setItemsStream method
+    And    several items are stored in the Selector instance
```
Comment (stephanebenayoun) [inline tests/features/service-selector.feature]: You should not assume what is stored or not and the test should not really care about the `selector instance`.
This steps can be described as 
```gherkin
Given ...
And user calls setItemsStream with stream of items A
When user calls setItemsStream with stream of items B
```

**Review by stephanebenayoun — COMMENTED** [review]
As a general comment, test cases should not include statement on what is stored or not by the selector instance. You should only care about calling methods and getting some results like any other API testing. 
You don't care about how the service is working and storing things, it's internal and depends on implementation, you just care about what happens when user call methods of the service in a specific context, and the context is also defined by calling methods of the service. 

For example :
```gherkin
Given service A
And user calls method X with parameter 1,2,3
When user calls to method Y with parameter 4,5,6
And user subscribe to method Z
Then the subscription to method Z emit items 4,5,6
```

**Review by stephanebenayoun — APPROVED** [review]

**Pull request merged**